drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -67,13 +82,24 @@ static int stm32_temp_channel_get(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+#if HAS_CALIBRATION
+	temp = ((float)data->raw * cfg->tsv_mv) / cfg->cal_vrefanalog;
+	temp -= *cfg->cal1_addr;
+	temp *= (cfg->cal2_temp - cfg->cal1_temp);
+	temp /= (*cfg->cal2_addr - *cfg->cal1_addr);
+	temp += cfg->cal_offset;
+#else
+	int32_t mv = data->raw * cfg->tsv_mv / 0x0FFF; /* Sensor value in millivolts */
+
 	if (cfg->is_ntc) {
-		temp = (float)(cfg->v25_mv - data->mv);
+		temp = (float)(cfg->v25_mv - mv);
 	} else {
-		temp = (float)(data->mv - cfg->v25_mv);
+		temp = (float)(mv - cfg->v25_mv);
 	}
-	temp = (temp/cfg->avgslope)*10;
+	temp = (temp / cfg->avgslope) * 10;
 	temp += 25;
```

Comment:
Is this formula correct? Shouldn't you add 30 instead?
![image](https://user-images.githubusercontent.com/1563974/149173153-d7d6ca84-91b9-4991-9812-93e144c5cc6a.png)